### ISSUE TITLE
Gensim broken with SciPy 1.13.0

### ISSUE THREAD

**Issue by jc-louis** [piskvorky/gensim#3525]
Gensim is broken with [SciPy 1.13.0](https://github.com/scipy/scipy/releases/tag/v1.13.0) which was released today.

This is due to the [removal of scipy.linalg.triu](https://github.com/scipy/scipy/pull/18021)

```
.venv/lib/python3.11/site-packages/gensim/__init__.py:11: in <module>
    from gensim import parsing, corpora, matutils, interfaces, models, similarities, utils  # noqa:F401
.venv/lib/python3.11/site-packages/gensim/corpora/__init__.py:6: in <module>
    from .indexedcorpus import IndexedCorpus  # noqa:F401 must appear before the other classes
.venv/lib/python3.11/site-packages/gensim/corpora/indexedcorpus.py:14: in <module>
    from gensim import interfaces, utils
.venv/lib/python3.11/site-packages/gensim/interfaces.py:19: in <module>
    from gensim import utils, matutils
.venv/lib/python3.11/site-packages/gensim/matutils.py:20: in <module>
    from scipy.linalg import get_blas_funcs, triu
E   ImportError: cannot import name 'triu' from 'scipy.linalg
```

**Comment by arsenetripard** [comment]
In the meantime this if fixed, if you're using Poetry, you can add to `pyproject.toml`:
```
scipy = "<1.11.0"
```

**Comment by anuoluwatuyi** [comment]
I was able to get this working by editing lines 20 to 26 in the `matutils.py` of the gensim library as implemented [here](https://github.com/piskvorky/gensim/commit/ad68ee3f105fc37cf8db333bfb837fe889ff74ac)


```
from scipy.linalg import get_blas_funcs
from scipy.linalg.lapack import get_lapack_funcs
from scipy.special import psi  # gamma function utils
try:
    from numpy import triu
except ImportError:
    from scipy.linalg import triu
```

**Comment by shreyasvinaya** [comment]
Hi, 
Could you please let me know when the next release of gensim would be
Thanks in Advance

**Comment by TikoyX** [comment]
When will this be released? :)
For others: Downgrade SciPy to version 1.12 to fix this issue

**Comment by evemilano** [comment]
It's just a boring thing when these things happen. maintaining backwards compatibility would be a good thing.

**Comment by piskvorky** [comment]
I agree… but the right place for that sigh would be the scipy board :)

**Comment by marianpascalau** [comment]
Hallo ... this is actually not an error in "SciPy"!
The problem is in "gensim" because is calling a method that do not exist anymore (it was removed and should be called from somewhere else).

This situation is documented and a warning was issued as well:
https://docs.scipy.org/doc/scipy/release/1.11.0-notes.html

**Comment by marianpascalau** [comment]
Just as a question... if this issue is closed does it means that it will not be fixed anymore?

**Comment by piskvorky** [comment]
This issue was closed because we merged a pull request that fixed it. Search above for `completed`.

**Comment by marianpascalau** [comment]
Hi piskvorky ... 
with all the respect, "you call this a fix?"

**Comment by anuoluwatuyi** [comment]
> Hi piskvorky ... 
> 
> with all the respect, "you call this a fix?"
> 
> 

It's been fixed @marianpascalau. 
It just hasn't been released yet. 
If you'd like, you could implement the fix in your gensim library ahead of the release.

**Comment by jeanmonet** [comment]
> > Hi piskvorky ...
> > with all the respect, "you call this a fix?"
> 
> It's been fixed @marianpascalau. It just hasn't been released yet. If you'd like, you could implement the fix in your gensim library ahead of the release.

```bash
pip install git+https://github.com/piskvorky/gensim.git
```

**Comment by bdpedigo** [comment]
> This issue was closed because we merged a pull request that fixed it. Search above for `completed`.
> 
> The fix has not been released yet though = it's only on Github, not on PyPI. So the advice above for a workaround still applies, unless you are installing gensim directly from Github.

Hi @piskvorky, do you have a timeline for when this issue might be fixed with a release on PyPI?

**Comment by mpenkov** [comment]
Hasn't this been fixed in https://pypi.org/project/gensim/4.3.3/, released several months ago?

**Comment by bdpedigo** [comment]
> Hasn't this been fixed in https://pypi.org/project/gensim/4.3.3/, released several months ago?

Thanks! My apologies somehow I missed that on PyPI

**Comment by ShailMurtaza** [comment]
Hi!
I'm facing the same issue. So pypi package is not up to date?
